src/watts/plugin_moose.py

```diff
@@ -167,8 +167,21 @@ def prerun(self, params: Parameters):
         ----------
         params
             Parameters used when rendering template
-
         """
+        # Perform unit conversion if necessary.
+        # Variables are converted to the specified units.
+        # If no units are specified, variables are converted to SI by default.
+        for key in params.keys():
+            if key in params.dict_unit_conv.keys() and key not in params.converted_param:
+
+                dict_param = params.dict_unit_conv[key]
+
+                if "new_unit" in dict_param.keys():
+                    params[key] = params.convert_units(dict_param=dict_param)
```

Comment:
One thing I think we can improve here is that the plugin classes are actually modifying the `params` that are getting passed in -- ideally that argument should be unchanged. An easy way around this would be for the plugin to make a copy of the `Parameters` instance before making any changes in it.